index.html

```diff
@@ -2543,18 +2588,269 @@ <h1>
+            </dl>
+        </p>
+        
+        
+        <p>
+The <a>DID resolver</a> executes the "Read" operation of the <a>DID method</a> as described in
+<a href="#read-verify"></a>. If successful, the outputs of this function MUST contain a <a>DID document</a>, 
+a set of metadata for the DID document (which MAY be empty), and a set of metadata for the resolution process. If
+the function results in an error, the outputs MUST contain a set of metadata for the resolution process.
+        </p>
+        
+        <p>
+The <a>DID document</a> is returned as a byte stream of a conformant representation
+as determined and supported by the <a>DID resolver</a>. This component is REQUIRED unless an error is returned
+by the <a>DID resolution</a> function. The client of the <a>DID resolution</a> function
+can then parse and process the <a>DID document</a> from this byte stream.
+        </p>
+        
+        <p>
+When a <a>DID document</a> is returned, the DID document metadata is returned as a map of
+key-value string pairs as described in <a href="#metadata-structure"></a>. This metadata 
+contains information about the input <a>DID</a> and the returned <a>DID document</a>. This metadata
+typically does not change between invocations of the <a>DID resolution</a> function. The keys and
+possible values for this metadata are defined by a registry (TBD). This component MAY be empty.
+        </p>
+
+        <p class="issue" data-number="203">
+The contents of the DID document metadata still needs to be defined within this document.
+        </p>
+
+        <p>
+The DID resolution metadata is returned as a map of key-value string pairs as described in
+<a href="#metadata-structure"></a>. This metadata contains information about the results of
+the resolution process. This metadata typically changes between invocations of the <a>DID resolution</a>
+function. The keys and possible values for this metadata are defined by a registry (TBD). This
+specification defines the following keys and values:
+
+            <dl>
+                <dt>content-type</dt>
+                <dd>The mime-type of the returned conformant representation of the <a>DID document</a> if successful.
+                    The <a>DID resolver</a> MUST supply this value when a <a>DID document</a>
+                    is returned. The <a>client</a> MUST use this value when determining how to
+                    parse and process the byte stream returned by this function into
+                    <a>DID document</a>.</dd>
+
+                <dt>error</dt>
+                <dd>The error code from the resolution process. The <a>DID resolver</a> MUST supply
+                    this value when there is an error. The possible values
+                    this field are defined by a registry (TBD). This specification defines the following
+                    error values:
+                    
+                    <dl>
+                        <dt>invalid-did</dt>
```

Comment:
I think it would be useful to include errors around network failure and other failures that occur within the resolution process that could be encompassed together. Something like a Resolution failure error.

Reply:
I agree but didn't want to get into that discussion in this pull request. That's for the WG to decide once the structure is in place. Right now there's not even a way to communicate errors to anything.

Reply:
Sounds good, makes sense to hold off on that until we get a first cut at it in.